src/app/clusters/ecosystem-information-server/ecosystem-information-server.cpp

```diff
@@ -251,6 +251,17 @@ EcosystemInformationServer & EcosystemInformationServer::Instance()
     return mInstance;
 }
 
+CHIP_ERROR EcosystemInformationServer::AddEcosystemInformationClusterToEndpoint(EndpointId aEndpoint)
+{
+    VerifyOrReturnError((aEndpoint != kRootEndpointId && aEndpoint != kInvalidEndpointId), CHIP_ERROR_INVALID_ARGUMENT);
+    auto it = mDevicesMap.find(aEndpoint);
+    // We expect that the device has not been previously added.
+    VerifyOrReturnError((it == mDevicesMap.end()), CHIP_ERROR_INCORRECT_STATE);
+    // This create an empty DeviceInfo in mDevicesMap.
+    mDevicesMap[aEndpoint];
```

Comment:
Even with the comment, this made me do a double take. Does mDeviceMap[aEndpoint] = DeviceInfo(); work? Do all the struct members auto-initialize to empty / nullopt?

Not for this PR, but there are a couple of heap-eater stds in there. Is this cluster restricted in some way to NOT be on embedded platforms?

Reply:
Yes that is explicitly stated where CI would normally block on allowing these types 

https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/check_includes_config.py#L131

Reply:
okie dokie - thanks. Sounds like it's covered.